src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -120,6 +124,24 @@ public object GetCursor(IStreamIdentity streamIdentity, StreamSequenceToken sequ
             return cursor;
         }
 
+        private void ReportCacheMessageStatistics(object state)
+        {
+            if (this.IsEmpty)
+            {
+                this.cacheMonitor.ReportMessageStatistics(null, null, null, this.ItemCount);
```

Comment:
nah, hardcode is not good. what if there's a bug in this.isEmpty , it returns true even when it is not empty?